internal/store/leaf/fsck.go

```diff
@@ -59,10 +62,14 @@ func (s *Store) fsckCheckEntry(ctx context.Context, name string) error {
 	// make sure we can actually decode this secret
 	// if this fails there is no way we could fix this
 	if IsFsckDecrypt(ctx) {
-		_, err := s.Get(ctx, name)
+		ctx = ctxutil.WithShowParsing(ctx, true)
+		secret, err := s.Get(ctx, name)
 		if err != nil {
 			return errors.Wrapf(err, "failed to decode secret %s: %s", name, err)
 		}
+		if secret.Type() == "Mime" {
```

Comment:
This should be: `if kv, ok := secret.(secrets.KV); ok && kv.FromMime() {`

See my other comment.